Python/sqlmlutils/sqlpythonexecutor.py

```diff
@@ -124,9 +121,13 @@ def create_sproc_from_function(self, name: str, func: Callable,
             input_params = {}
         if output_params is None:
             output_params = {}
+
+        in_copy = input_params.copy() if input_params is not None else None
+        out_copy = output_params.copy() if output_params is not None else None

```

Comment:
why do we need to copy?

Reply:
We modify input_params/output_params when creating the stored procedure because we need to add stdout and stderr as params. If we don't copy the dictionaries then we actually modify the underlying contents.

I'll add a comment here explaining this